sdcm/provision/azure/utils.py

```diff
@@ -72,3 +73,13 @@ def get_scylla_images(  # pylint: disable=too-many-branches
     if only_latest:
         return output[-1:]
     return output
+
+
+IMAGE_URL_REGEX = re.compile(
+    r'.*/resourceGroups/(?P<resource_group_name>.*)/providers/Microsoft.Compute/images/(?P<image_name>.*)')
+
+
+def get_image_tags(link: str) -> dict:
```

Comment:
for Azure there's no need to do anything as user-data taken by SMI and cloud-init input are separated.
cloud-init script is provided by `custom_data` in `os_profile`, while SMI takes content of `user_data` which is stored in metadata. `AzureProvider` understands that and so other providers in unified provisioning will have to understand the difference between SMI's `user_data` and cloud-init's `user_data`.

Reply:
The idea is to align the way we provide user-data, and no have a different way for each cloud, like we have now.

The common thing between all of them is user-data, and yes we should be using cloud-init for most of the configuration.

and we should be aware of what SMI supports also for azure image